src/main/java/MegaMillions.java

```diff
@@ -0,0 +1,83 @@
+import java.io.*;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MegaMillions {
+
+    File file = new File("src/main/resources/mega_millions.csv");
+    String absolutePath = file.getAbsolutePath();
+
+    public void filterAndPrintNumbers() {
+        List<Integer> allWinningNumbers = new ArrayList<>();
+        List<Integer> allWinningMegaBall = new ArrayList<>();
+        getAllWinningNumbers(allWinningNumbers, allWinningMegaBall);
+        List<Number> winning75Numbers = getUniqueNumbersWithOccurs(allWinningNumbers);
+        List<Number> winningMegaBalls = getUniqueNumbersWithOccurs(allWinningMegaBall);
+        System.out.println("Top 10 winning numbers:");
+        mostPopularNumberFromList(winning75Numbers, 10);
+        System.out.println("Top 3 winning mega ball:");
+        mostPopularNumberFromList(winningMegaBalls, 3);
+    }
+
+    private void getAllWinningNumbers(List<Integer> winningNumbers, List<Integer> megaBalls) {
+        String line = "";
+        try (
+                var fileReader = new FileReader(absolutePath);
+                var bufferedReader = new BufferedReader(fileReader);
+        ) {
+            while ((line = bufferedReader.readLine()) != null) {
+                addWinningNumbers(line, winningNumbers);
+                addMegaBalls(line,megaBalls);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void addWinningNumbers(String line, List<Integer> winningNumbersList) {
+        String[] splitByComma = line.split(",");
+        String[] numbers = splitByComma[1].split(" ");
+        for (String number : numbers) {
+            if (number.matches("-?\\d+")) {
+                winningNumbersList.add(Integer.parseInt(number));
+            }
+        }
+    }
+
+    public void addMegaBalls(String line, List<Integer> megaBalls){
+        String[] splitByComma = line.split(",");
+        String[] megaBall = splitByComma[2].split(" ");
+        for (String ball : megaBall) {
+            if (ball.matches("-?\\d+")) {
```

Comment:
Are you sure this regex works well?

Reply:
I couldn't find regex "-?" but without "-" code doesn't compile.  
? - Match the previous element zero or one times (one if possible)
d - Matches digits (same as [0-9])
+ - Match the previous element one or more times (as many as possible)
http://regexrenamer.sourceforge.net/help/regex_quickref.html
I'm doesn't understand it at all but code is workind.

Reply:
I just asked because regex is tricky but it seems you've done your research